src/aggregate/aggregate_request.c

```diff
@@ -1395,15 +1395,15 @@ void AREQ_Free(AREQ *req) {
 
   ConcurrentSearchCtx_Free(&req->conc);
 
-  // Finally, free the context. If we are a cursor, some more
-  // cleanup is required since we also now own the
-  // detached ("Thread Safe") context.
+  // Finally, free the context. If we are a cursor or have multi workers threads,
+  // we need also to detach the ("Thread Safe") context.
   RedisModuleCtx *thctx = NULL;
   if (req->sctx) {
-    if (req->reqflags & QEXEC_F_IS_CURSOR) {
+    if (req->reqflags & QEXEC_F_HAS_THCTX || req->reqflags & QEXEC_F_IS_CURSOR) {

```

Comment:
The flag `QEXEC_F_HAS_THCTX` is set also for cursor
```
if ((*r)->reqflags & QEXEC_F_IS_CURSOR) {
  ...
  (*r)->reqflags |= QEXEC_F_HAS_THCTX;
  ...
}
```
So here no need to also test `QEXEC_F_IS_CURSOR`?

Reply:
The coordinator for example also uses a cursor and sets `QEXEC_F_IS_CURSOR` flag on.
for backward compatibility i check both

Reply:
Seemed a bit nicer that the code here could care/know less, and just handle the flag `QEXEC_F_HAS_THCTX`
And wherever a thread safe context is set, such as by the coordinator, the flag would be set as well.

Reply:
I don't think it's in the scope of this PR to change the current behaviour of the existing code (in the coordinator for example we set the cursor flag and get the threadsafe context somewhere else)
we can do it in a different PR :)